assembly/src/tokens/stream.rs

```diff
@@ -21,15 +28,34 @@ impl<'a> TokenStream<'a> {
         if source.is_empty() {
             return Err(AssemblyError::empty_source());
         }
+        let mut doc_comments = BTreeMap::new();
+
+        let mut tokens = Vec::new();
+        let mut comment = Comment(String::new());
 
-        let tokens = source
-            .lines()
-            // Tokenize and remove comments
-            .flat_map(|line| {
-                line.split_whitespace()
+        let lines = source.lines();
+        for line in lines {
```

Comment:
Is there a reason not to do just:
```Rust
for line in source.lines() {

}
```
Doesn't seem like we need access to `lines` variable later.